src/ent/board.h

```diff
@@ -3,157 +3,462 @@
+                                       int cross_index, uint64_t cross_set) {
+  grid_set_cross_set(board_get_mutable_grid(b, b->transposed), row, col, dir,
+                     cross_index, cross_set);
+  grid_set_cross_set(board_get_mutable_grid(b, 1 - b->transposed), col, row,
+                     1 - dir, cross_index, cross_set);
+}
+
+static inline void board_set_cross_set_letter(Board *b, int row, int col,
+                                              int dir, int cross_index,
                                               uint8_t letter) {
-  *cross_set = *cross_set | ((uint64_t)1 << letter);
+  grid_set_cross_set_letter(board_get_mutable_grid(b, b->transposed), row, col,
+                            dir, cross_index, letter);
+  grid_set_cross_set_letter(board_get_mutable_grid(b, 1 - b->transposed), col,
+                            row, 1 - dir, cross_index, letter);
+}
+
+// Board: Cross score
+
+static inline int board_get_cross_score(const Board *b, int row, int col,
+                                        int dir, int cross_index) {
+  return grid_get_cross_score(board_get_const_grid(b, b->transposed), row, col,
+                              dir, cross_index);
+}
+
+static inline void board_set_cross_score(Board *b, int row, int col, int dir,
+                                         int cross_index, int cross_score) {
+  grid_set_cross_score(board_get_mutable_grid(b, b->transposed), row, col, dir,
+                       cross_index, cross_score);
+  grid_set_cross_score(board_get_mutable_grid(b, 1 - b->transposed), col, row,
+                       1 - dir, cross_index, cross_score);
 }
 
-static inline void board_set_cross_set(Board *board, int row, int col,
-                                       uint64_t letter, int dir,
-                                       int cross_set_index) {
-  board->cross_sets[board_get_tindex_player_cross(board, row, col, dir,
-                                                  cross_set_index)] = letter;
+// Board: Anchors
+
+static inline bool board_get_anchor(const Board *b, int row, int col, int dir) {
+  return grid_get_anchor(board_get_const_grid(b, b->transposed), row, col, dir);
+}
+
+static inline void board_set_anchor(Board *b, int row, int col, int dir,
+                                    bool anchor) {
+  grid_set_anchor(board_get_mutable_grid(b, b->transposed), row, col, dir,
+                  anchor);
+  grid_set_anchor(board_get_mutable_grid(b, 1 - b->transposed), col, row,
+                  1 - dir, anchor);
+}
+
+static inline void board_reset_anchor(Board *b, int row, int col, int dir) {
+  grid_reset_anchor(board_get_mutable_grid(b, b->transposed), row, col, dir);
+  grid_reset_anchor(board_get_mutable_grid(b, 1 - b->transposed), col, row,
+                    1 - dir);
+}
+
+static inline void board_reset_number_of_anchor_rows(Board *b) {
+  for (int i = 0; i < 2; i++) {
+    Grid *g = board_get_mutable_grid(b, i);
+    for (int i = 0; i < BOARD_DIM; i++) {
```

Comment:
please ignore me if you fixed this already but j instead of i for the inner loop would be more readable. (don't think it has any actually effect because it uses the i at the right scope)

Reply:
oops, didn't mean to do that, I'll fix that. I should have some compiler warnings for variable shadowing.